potts_model.py

```diff
@@ -291,6 +292,35 @@ def coupling_scale(self):
   def field_scale(self):
     return self._field_scale
 
+  @property
+  @functools.lru_cache()
+  def epistasis_tensor(self):
+      """Returns the epistasis tensor with respect to the wildtype sequence
+
+      Recall that epistasis is given by:
+      $$
+          \epsilon_{i \beta, j \gamma} =
+          H_{i \beta, j \gamma}
+          - H_{i a, j \gamma} - H_{i \beta, j a}
+          + H_{i a, j a}
+      $$
+      """
+      H = self.weight_matrix
+      L = H.shape[0]
+      A = H.shape[2]
+      epistasis_tensor = np.zeros_like(H)
+
+      for i in range(L):
```

Comment:
this could surely be vectorized. If this becomes a computational bottleneck, let's talk about speeding it up.